File: tests/benchmarks/resolution/fixtures/lua/validators.lua
local M = {}

local function check_not_empty(value)
    return value ~= nil and value ~= ""
end

function M.validate_name(name)
    if not check_not_empty(name) then
        return false, "name must not be empty"
    end
    if #name < 2 then
        return false, "name must be at least 2 characters"
    end
    return true, nil
end

function M.validate_email(email)
    if not check_not_empty(email) then
        return false, "email must not be empty"
    end
    if not string.find(email, "@") then
        return false, "email must contain @"
    end
    return true, nil
end

return M

File: tests/benchmarks/resolution/fixtures/r/validators.R
validate_email <- function(email) {
  if (!grepl("@", email)) {
    stop("Invalid email address")
  }
  TRUE
}

validate_name <- function(name) {
  if (nchar(name) < 2) {
    stop("Name must be at least 2 characters")
  }
  TRUE
}

validate_user_input <- function(name, email) {
  validate_name(name)
  validate_email(email)
}
